app/src/org/commcare/CommCareApplication.java

```diff
@@ -1153,10 +1155,18 @@ public ModernHttpRequester buildHttpRequester(Context context, String url,
                                                   AuthInfo authInfo,
                                                   @Nullable HttpResponseProcessor responseProcessor, boolean retry) {
 
-        CommCareNetworkService networkService;
+        CommCareNetworkService networkService = null;
         if (authInfo instanceof AuthInfo.NoAuth) {
             networkService = CommCareNetworkServiceGenerator.createNoAuthCommCareNetworkService();
-        } else {
+        } else if(authInfo instanceof AuthInfo.CurrentAuth) {
+            //Try to get SSO token
+            AuthInfo.TokenAuth tokenAuth = ConnectIDSSOHelper.acquireSSOTokenSync(context);

```

Comment:
should this only happen if connect ID is enabled ?

Reply:
I added a check at the top of acquireSSOTokenSync() so it returns null if ConnectID is locked.